include/realizations/catchment/Catchment_Formulation.hpp

```diff
@@ -7,7 +7,8 @@
 #include "Et_Accountable.hpp"
 #include <HY_CatchmentArea.hpp>
 
-#include <Forcing.h> // Remove when _link_legacy_forcing() is removed!
+//#include <Forcing.h> // Remove when _link_legacy_forcing() is removed!
+#include "CsvPerFeatureForcingProvider.hpp"
```

Comment:
Some of the time functions, get_data_start_time() and several others, for example are only provided in the following header files, so we need to provide CsvPerFeatureForcingProvider some where. I just tried to use GenericDataProvider.hpp and DataProvider.hpp in its place, both gave build errors.
```
include/forcing/CsvPerFeatureForcingProvider.hpp:    long get_data_start_time() override {
include/forcing/DataProvider.hpp:        virtual long get_data_start_time() = 0;
include/forcing/WrappedForcingProvider.hpp:            return wrapped_provider->get_data_start_time();
include/forcing/WrappedForcingProvider.hpp:        long get_data_start_time() override {
include/forcing/WrappedForcingProvider.hpp:            return wrapped_provider->get_data_start_time();
include/forcing/Forcing.h:    long get_data_start_time() override
include/forcing/NetCDFPerFeatureDataProvider.hpp:        long get_data_start_time()
```

Reply:
Confused on this, since as you show above, `get_data_start_time()` is defined in both, and it's not even templated... will try this myself and have a look.

Reply:
It seems to me the `get_data_start_time() `is only accessible through `Forcing, CsvPerFeatureForcingProvider, WrappedForcingProvider, or NetCDFPerFeatureDataProvider`.

Reply:
You maybe right. Probably need some code reorganization.

Reply:
The issue here doesn't seem to have anything to do with `get_data_start_time()`. See the comment in `Formulation_Constructors.hpp`.